### DISCUSSION TITLE
autoexec.ipxe with IPv6

### DISCUSSION THREAD

**Discussion by NiKiZe** [ipxe/ipxe#1355] [General]
Observed behaviour 
When booting in UEFI with IPv6 no IP is seen in the boot path, no cwurl can be created and no autoexec.ipxe attempt is done.

Is this just a boot path thing in the firmware I have used, or is it something that maybe haven't been tested?

Could the firmware issues somehow be mitigated if dhcpcache supported dhcpv6?

I would like to have given more details and tested more, but my environment just doesn't want to play nice today.

This was seen while documenting IPv6 boot, notes at
https://gist.github.com/NiKiZe/3da6bbc9bd54c79645d833b9ed271ef9

**Comment by NiKiZe** [comment]
Uefi does in these cases not seem to include bootfile-url or any other settings (everything is 0) in the uefi path.
But that is true for IPv4 as well, I fail to see the full flow here leading to IPv4 working, I still assume this has to do with the dhcpcache?

iPXE does have `DHCPV6_BOOTFILE_URL` (option 59) but fails to give me any reasonable value - but that is probably more likely due to IPv4 proxy dhcp.

**Comment by jip149** [comment]
I ran into the tame problem. Should an issue be opened? Is there any way to fix this?

**Reply by NiKiZe** [reply]
If the issue is with the EFI firmware, then it needs to be resolved there first, which means EDK2.
There is no point opening an issue in iPXE until all datapoints is there for what is missing, this discussion serves as the point to collect that data.

Let's start with building with `DEBUG=efi_autoexec,efi_path` try to grab both IPv6 and IPv4 boot on the same device to compare. Maybe stating the obvious but in both cases it must be chained over dhcp+tftp, http boot over IPv6 is then another interesting part of this.

But with that said, it reminds me of reported shim issues, even if there is no path found, iPXE should/could resort to trying to just fetch autoexec.ipxe ? (another path to investigate)

**Reply by jip149** [reply]
Thanks, I will try that and report back.

**Comment by jip149** [comment]
dhcp4:

<img width="1920" height="1080" alt="mpv-shot0009" src="https://github.com/user-attachments/assets/8c8b82d2-205b-4f62-b4a8-e2fdd0cb01cd" />

dhcp6:
<img width="1920" height="1080" alt="mpv-shot0010" src="https://github.com/user-attachments/assets/efe89210-9f91-4acc-b76f-3ce8fc8983cf" />
<img width="1920" height="1080" alt="mpv-shot0011" src="https://github.com/user-attachments/assets/be106704-920c-4e69-977c-80b3a8ed89d7" />
<img width="1920" height="1080" alt="mpv-shot0012" src="https://github.com/user-attachments/assets/07e110cb-f330-45f8-a725-cae566184774" />

dhcp6, with following script embedded:
```
#!ipxe

dhcp -c ipv6
chain --replace --autofree tftp://[2001:912:1480:310:2400::1]/autoexec.ipxe
```

<img width="1920" height="1080" alt="mpv-shot0015" src="https://github.com/user-attachments/assets/b7754da1-ebea-4cf8-abf8-725623f6cec6" />

Note: this is real hardware, not a VM.

**Reply by NiKiZe** [reply]
Created #1703
